recipe/magento2.php

```diff
@@ -219,3 +221,137 @@
+set('artifact_path', function () {
+    if (!test('[ -d {{artifact_dir}} ]')) {
+        run('mkdir {{artifact_dir}}');
+    }
+    return get('artifact_dir') . '/' . get('artifact_file');
+});
+
+set('bin/tar', function () {
+    if (commandExist('gtar')) {
+        return which('gtar');
+    } else {
+        return which('tar');
+    }
+});
+
+set('cacheToolPath', function() {
+    return get('cacheTool', '{{current_path}}/bin/cachetool');
+});
+
+// tasks section
+desc('Packages all relevant files in an artifact.');
+task('artifact:package', function() {
+    if (!test('[ -f {{artifact_excludes_file}} ]')) {
+        throw new GracefulShutdownException(
+            "No artifact excludes file provided, provide one at artivacts/excludes or change location"
+        );
+    }
+    run('{{bin/tar}} --exclude-from={{artifact_excludes_file}} -czf {{artifact_path}} .');
+});
+
+desc('Uploads artifact in release folder for extraction.');
+task('artifact:upload', function () {
+    upload(get('artifact_path'), '{{release_path}}');
+});
+
+desc('Extracts artifact in release path.');
+task('artifact:extract', function () {
+    run('{{bin/tar}} -xzpf {{release_path}}/{{artifact_file}} -C {{release_path}}');
+    run('rm -rf {{release_path}}/{{artifact_file}}');
+});
+
+desc('Provides env.php for build.');
+task('build:prepare-env', function() {
+    $deployEnv = get('deploy_env','app/etc/deploy.php');
+    if (!test('[ -f ./'.$deployEnv.' ]')) {
+        throw new GracefulShutdownException(
+            "No deploy env provided, provide one at app/etc/deploy.php or change location"
+        );
+    }
+    run ('cp '.$deployEnv.' app/etc/env.php');
+});
+
+desc('Clears generated files prior to building.');
+task('build:remove-generated', function() {
+    run('rm -rf generated/*');
+});
+
+desc('Clears the opcache, cache tool required.');
+task('cache:clear:opcache', function() {
+    if ($fpmSocket = get('fpm_socket', '')) {
+        run('{{bin/php}} -f {{cacheToolPath}} opcache:reset --fcgi '.$fpmSocket);
+    }
+});
+
+desc('Builds an artifact.');
+task('artifact:build', function () {
+    if(currentHost()->get('local')) {
+        set('deploy_path', '.');
+        set('release_path', '.');
+        set('current_path', '.');
+        invoke('build:prepare-env');
+        invoke('build:remove-generated');
+        invoke('deploy:vendors');
+        invoke('magento:compile');
+        invoke('magento:deploy:assets');
+        invoke('artifact:package');
+    } else {
+        throw new GracefulShutdownException("Artifact can only be built locally, you provided a non local host");
+    }
+});
+
+desc('Prepares an artifact on the target server');
+task('artifact:prepare', function(){
+    if(currentHost()->get('local')) {
+        throw new GracefulShutdownException("You can only deploy to a non localhost");
+    } else {
+        add('shared_files', get('additional_shared_files') ?? []);
+        add('shared_dirs', get('additional_shared_dirs') ?? []);
```

Comment:
Not quite sure if I get it. `'shared_files'` and `'shared_dirs'` are defined in the standard magento2 recipe. So I think I should not comment these. and the  `'additional_shared_files'` and `'additional_shared_dirs'` are just provided to add some nonstandard files or dirs coming from your shop.
So anyhow I can comment this, but how would I do it, since the documentation is created :thinking:

Reply:
The documentation is created from comments before `set()` calls. I'll do it in our fork and show you

Other changes I already implemented: https://github.com/integer-net/deployer/pulls?q=is%3Apr+

Reply:
See here: https://github.com/integer-net/deployer/pull/11/files

Reply:
BTW: I had to move the `add('...`-part in a task (`depoy:additional-shared`), just like in your PR it did not add the yml settings, I guess because of timing.
```
desc('Adds additional files and dirs to the list of shared files and dirs');
task('deploy:additional-shared', function () {
    add('shared_files', get('additional_shared_files'));
    add('shared_dirs', get('additional_shared_dirs'));
});
```